octobot_trading/orders/order_state.py

```diff
@@ -0,0 +1,184 @@
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+
+from octobot_commons.logging.logging_util import get_logger
+
+from octobot_trading.channels.exchange_channel import get_chan
+from octobot_trading.constants import ORDERS_CHANNEL
+from octobot_trading.enums import OrderStates
+from octobot_trading.util.initializable import Initializable
+
+
+class OrderState(Initializable):
+    DEFAULT_SYNC_RETRY_TIMEOUT = 5
+
+    def __init__(self, order, is_from_exchange_data):
+        super().__init__()
+
+        # related order
+        self.order = order
+
+        # default state
+        self.state = OrderStates.UNKNOWN
+
+        # if this state has been created from exchange data or OctoBot internal mechanism
+        self.is_from_exchange_data = is_from_exchange_data
+
+        # the updating task
+        self.updating_task = None
+
+    def is_pending(self) -> bool:
+        """
+        :return: True if the state is pending for update
+        """
+        return self.state is OrderStates.UNKNOWN
+
+    def is_refreshing(self) -> bool:
+        """
+        :return: True if the state is updating
+        """
+        return self.state is OrderStates.REFRESHING
+
+    def is_open(self) -> bool:
+        """
+        :return: True if the Order is considered as open
+        """
+        return not (self.is_filled() or self.is_canceled() or self.is_closed())
+
+    def is_filled(self) -> bool:
+        """
+        :return: True if the Order is considered as filled
+        """
+        return False
+
+    def is_closed(self) -> bool:
+        """
+        :return: True if the Order is considered as closed
+        """
+        return False
+
+    def is_canceled(self) -> bool:
+        """
+        :return: True if the Order is considered as canceled
+        """
+        return False
+
+    def get_logger(self):
+        """
+        :return: the order logger
+        """
+        return get_logger(self.order.get_logger_name())
+
+    def log_order_event_message(self, state_message):
+        """
+        Log an order state event
+        """
+        self.get_logger().info(f"{self.order.symbol} {self.order.get_name()} at {self.order.origin_price}"
```

Comment:
We should use the order str() representation here instead of ordre.get_XYZ(), what do you think?

Reply:
Good idea :+1: